src/audio/drc/drc.c

```diff
@@ -261,6 +298,9 @@ static int drc_process(struct processing_module *mod,
 		}
 	}
 
+	/* Control pass-though in processing function with switch control */
+	cd->enabled = drc_get_config_enabled(cd) && cd->enable_switch;
```

Comment:
better to directly as:
cd->enabled = cd->config && cd->config->params.enabled
                         && cd->enable_switch;